iac-extensions/spring-config/src/main/java/org/sonar/iac/springconfig/parser/yaml/SpringConfigYamlConverter.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.springconfig.parser.yaml;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+import org.snakeyaml.engine.v2.nodes.MappingNode;
+import org.snakeyaml.engine.v2.nodes.Node;
+import org.snakeyaml.engine.v2.nodes.NodeTuple;
+import org.snakeyaml.engine.v2.nodes.ScalarNode;
+import org.snakeyaml.engine.v2.nodes.SequenceNode;
+import org.sonar.iac.common.api.tree.Comment;
+import org.sonar.iac.common.api.tree.impl.TextRange;
+import org.sonar.iac.common.extension.ParseException;
+import org.sonar.iac.common.yaml.IacYamlConverter;
+import org.sonar.iac.common.yaml.tree.YamlTreeMetadata;
+import org.sonar.iac.springconfig.tree.api.File;
+import org.sonar.iac.springconfig.tree.api.Profile;
+import org.sonar.iac.springconfig.tree.api.Tuple;
+import org.sonar.iac.springconfig.tree.impl.FileImpl;
+import org.sonar.iac.springconfig.tree.impl.ProfileImpl;
+import org.sonar.iac.springconfig.tree.impl.ScalarImpl;
+import org.sonar.iac.springconfig.tree.impl.SyntaxTokenImpl;
+import org.sonar.iac.springconfig.tree.impl.TupleImpl;
+
+import static org.sonar.iac.springconfig.parser.SpringConfigProfileNameUtil.profileName;
+
+public class SpringConfigYamlConverter implements IacYamlConverter<File, Stream<SpringConfigYamlConverter.TupleBuilder>> {
+
+  private List<Comment> commentsPerProfile = new ArrayList<>();
```

Comment:
I chose to make the comment list stateful, as it did provide the least complex implementation IMO